apps/etterlatte-pdltjenester/src/main/kotlin/pdl/mapper/FamilieRelasjonMapper.kt

```diff
@@ -1,61 +1,159 @@
+            when (personRolle) {
+                PersonRolle.BARN ->
+                    hentPerson.foreldreansvar
+                        ?.filter { it.ansvarlig != null }
+                        ?.groupBy { it.ansvarlig }
+                        ?.mapValues { it.value.maxByOrNull { fa -> fa.metadata.sisteRegistrertDato() } }
+                        ?.mapNotNull {
+                            it.value?.ansvarlig?.let { Folkeregisteridentifikator.of(it) }
+                        }
+
+                else -> null
+            }
+        val foreldre =
+            when (personRolle) {
+                PersonRolle.BARN ->
+                    hentPerson.forelderBarnRelasjon
+                        ?.filter { it.relatertPersonsRolle != PdlForelderBarnRelasjonRolle.BARN }
+                        ?.groupBy { it.relatertPersonsIdent }
+                        ?.mapValues { it.value.maxByOrNull { fbr -> fbr.metadata.sisteRegistrertDato() } }
+                        ?.mapNotNull {
+                            it.value?.relatertPersonsIdent?.let { Folkeregisteridentifikator.of(it) }
+                        }
+
+                else -> null
+            }
+        val barn =
+            when (personRolle) {
+                PersonRolle.AVDOED, PersonRolle.GJENLEVENDE ->
+                    hentPerson.forelderBarnRelasjon
+                        ?.filter { it.relatertPersonsRolle == PdlForelderBarnRelasjonRolle.BARN }
+                        ?.groupBy { it.relatertPersonsIdent }
+                        ?.mapValues { it.value.maxByOrNull { fbr -> fbr.metadata.sisteRegistrertDato() } }
+                        ?.mapNotNull {
+                            it.value?.relatertPersonsIdent?.let { Folkeregisteridentifikator.of(it) }
+                        }
+
+                else -> null
+            }
+
+        val personerUtenIdent = mapPersonerUtenIdenter(hentPerson, personRolle)
+
+        if (personerUtenIdent.isNullOrEmpty()) {
+            return FamilieRelasjon(ansvarligeForeldre, foreldre, barn)
+        }
+
+        if (!aksepterPersonerUtenIdent) {
+            logger.error(
+                "Feilet i mapping av familierelasjon på grunn av personer som mangler identer. " +
+                    "Se sikkerlogg for detaljer.",
+            )
+            sikkerLogg.error(
+                "Feilet i mapping av familierelasjon på grunn av personer som mangler identer. " +
+                    "Mappet følgende personer som mangler identer: $personerUtenIdent",
+            )
+            throw IngenIdentFamilierelasjonException()
+        }
+
+        return FamilieRelasjon(
+            ansvarligeForeldre = ansvarligeForeldre,
+            foreldre = foreldre,
+            barn = barn,
+            personerUtenIdent = personerUtenIdent,
         )
+    }
+
+    private fun mapPersonerUtenIdenter(
+        hentPerson: PdlHentPerson,
+        personRolle: PersonRolle,
+    ): List<PersonUtenIdent>? {
+        val forelderBarnUtenIdent =
```

Comment:
Kunne vi splittet opp denne og foreldreUtenIdent i to egne funksjoner, synes denne funksjonen er relativt tung å lese.